Broken tests

lelutin opened the pull request:
the current code has a couple of tests that are failing. some maintenance is needed to make them pass.

most test failures were caused by the addition of some date formats -- meaning that we're now getting better parsed dates :100:  but the tests were not reflecting that.

tests for google.com.tr were failing because date parsing was apparently never implemented for that TLD.

a problem with the .ro parser that was introduced by the latest changes for consistency of data structure was exposed by flake8 so I fixed that as well.

This series also configures CI via github workflows to run tests on each push and pull request.

It also configures the CI to run flake8 in order to enforce a sound coding style. All of the flake8 errors are fixed by the last couple of commits.

richardpenman commented on whois/__init__.py:
```diff
@@ -1,20 +1,21 @@
-from .whois import NICClient
-import logging
+from builtins import *  # noqa
+import re  # noqa: E402
```
why are these lint comments needed here?

lelutin replied:
wow I've been asleep for so long.. should've followed up here but I suddenly didn't have enough energy time and focus.

for those lines specifically, I get those errors:

~~~
whois/__init__.py:9:1: F403 'from builtins import *' used; unable to detect undefined names
whois/__init__.py:9:1: F401 'builtins.*' imported but unused
whois/__init__.py:9:1: E402 module level import not at top of file
~~~

flake8 usually complains about `import *` but maybe we can disable more specific errors for the first line. in the other comment about the `import *` line, do you mean that we should rather explicitly import the parts of "builtins" that are needed instead of using "*" ?

for the second line, I guess it's complaining because on line 8 there's a call to `standard_library.install_aliases()`, so something gets executed before other imports. I've just tested moving that call down below all of the other import lines and the library still seems to be working properly while the flake8 issue is gone.
do you remember off-hand if there was a reason for the call to be placed above the other imports? by tracing the file with git I get to the commit "import from mercurial" so I can't see why the line was placed there.
if moving the call does not seem to be problematic, then we can do this and remove all of the "noqa: E402" comments.

richardpenman commented on whois/time_zones.py:
```diff
@@ -1,10 +1,13 @@
 from future import standard_library
 standard_library.install_aliases()
-from builtins import *
+from builtins import *  # noqa
```
instead of suppressing the lint warning the modules needed should be imported

lelutin commented:
I've just pushed a new merger with master, followed with lint fixes for what changed on master. We still have to figure out a best way forward wrt the discussions above before this can be merged to master

lelutin commented:
I sent some new test fixes.

I got quite confused by some results that were not behaving the same locally than on CI.

I just tried something and I think I've identified the source of the issue (but not yet how to get consistent output in all cases): when I have python-dateutil installed locally, many of the parsed dates don't contain the timezone indicators (e.g. "CLST", "+2:00") whereas when python-dateutil is not installed they get included in the results.  I'm not sure yet how to fix this situation.

FWIW, the tests are currently passing without python-dateutil

lelutin commented:
Well, I moved lots of things today on this branch. I've made it so that tests would verify what is output for `name_servers` since it's an important information. Then to fix tests with regards to name_servers, I've moved a bunch of TLD parsers to splitting name server items into a list instead of coughing up the whole matched string as-is to users.

In order to resolve the comments you had previously on the "noqa" bits on import lines, I've removed imports of builtins. They are not required in python 3.x. This probably means that python 2.x is now out the door if this branch gets merged. But 2.x has been EOL for a little while now.

And finally, CI now runs tests against python 3.10 as well.

If you are OK with the removal of the "import builtins", I think this should be ready for merge.  ..although it's grown into something pretty large, sorry about this -- I know it makes reviewing the changes harder :\

richardpenman commented:
Thanks for your work here - I have fixed all tests and migrated to pep-8 with black. Also added CI checks to make sure unit tests and flake8 passes for future commits.
